go/stellar/remote/remote.go

```diff
@@ -494,110 +183,103 @@ func FetchV2BundleForAccount(ctx context.Context, g *libkb.GlobalContext, accoun
+// all of the secrets. Also returned is a map of accountID->pukGen. Entries are only in the
+// map for accounts with secrets in the bundle. Inaccessible accounts will be in the
+// visible part of the parent bundle but not in the AccountBundle secrets nor in the
+// AccountPukGens map. FetchBundleWithGens is only for very specific usecases.
+// FetchAccountBundle and FetchSecretlessBundle are the preferred ways to pull a bundle.
+func FetchBundleWithGens(mctx libkb.MetaContext) (b *stellar1.Bundle, pukGen keybase1.PerUserKeyGeneration, accountGens bundle.AccountPukGens, err error) {
+	defer mctx.CTraceTimed("Stellar.FetchBundleWithGens", func() error { return err })()
+
+	b, _, pukGen, _, err = fetchBundleForAccount(mctx, nil) // this bundle no account secrets
+	if err != nil {
+		return nil, 0, bundle.AccountPukGens{}, err
+	}
+	accountGens = make(bundle.AccountPukGens)
+	newAccBundles := make(map[stellar1.AccountID]stellar1.AccountBundle)
+	for _, acct := range b.Accounts {
+		singleBundle, _, _, singleAccountGens, err := fetchBundleForAccount(mctx, &acct.AccountID)
 		if err != nil {
-			return nil, 0, 0, err
+			// expected errors include SCStellarDeviceNotMobile, SCStellarMobileOnlyPurgatory
+			mctx.CDebugf("unable to pull secrets for account %v which is not necessarily a problem %v", acct.AccountID, err)
+			continue
 		}
-		// strip the signers out of the bundle
-		newAccountBundles := make(map[stellar1.AccountID]stellar1.AccountBundle)
-		for accountID, ab := range acctBundle.AccountBundles {
-			newAb := ab.DeepCopy()
-			newAb.Signers = nil
-			newAccountBundles[accountID] = newAb
-		}
-		acctBundle.AccountBundles = newAccountBundles
+		accBundle := singleBundle.AccountBundles[acct.AccountID]
+		newAccBundles[acct.AccountID] = accBundle
+		accountGens[acct.AccountID] = singleAccountGens[acct.AccountID]
+	}
+	b.AccountBundles = newAccBundles
+	err = b.CheckInvariants()
+	if err != nil {
+		return nil, 0, bundle.AccountPukGens{}, err
 	}
-	return acctBundle, version, pukGen, err
+
+	return b, pukGen, accountGens, nil
 }
 
 // FetchWholeBundle gets the secretless bundle and loops through the accountIDs
 // to get the signers for each of them and build a single, full bundle with all
 // of the information. This will error from any device that does not have access
-// to all of the accounts (e.g. a desktop after mobile-only)
-func FetchWholeBundle(ctx context.Context, g *libkb.GlobalContext) (acctBundle *stellar1.BundleRestricted, version stellar1.BundleVersion, pukGen keybase1.PerUserKeyGeneration, err error) {
-	defer g.CTraceTimed(ctx, "Stellar.FetchWholeBundle", func() error { return err })()
-
-	mctx := libkb.NewMetaContext(ctx, g)
-	bundle, version, pukGen, err := FetchSecretlessBundle(mctx)
+// to all of the accounts (e.g. a desktop after mobile-only). This is only used
+// in tests.
+func FetchWholeBundle(mctx libkb.MetaContext) (bundle *stellar1.Bundle, err error) {
```

Comment:
maybe we could rename it `FetchWholeBundleForTesting`, assuming it won't work to shove it in an _test.go file.

Reply:
yeah - this was also used in Upkeep (which i've since replaced) and the CLI dump command (which only works in devel and has also since been replaced). i kinda kept chipping away at it because it seems less safe than fetching by account. but yeah. you're probably right. i think i can get this out of here.